src/re/multi_grid/grid.py

```diff
@@ -911,9 +784,9 @@ def toFlatGridAtLevel(self):
 
 @dataclass()
 class SparseGrid(FlatGrid):
```

Comment:
Should we stage the Sparse and Flat grids in this PR? In their current form and with resort gone, I believe they are not functional with kernels. Maybe it makes sense to split the features here?

Reply:
We need `FlatGrid` for compressing (optimizing) the kernel and I think it belongs into this PR.

I think we might be able to get rid of `resort` by virtue of `raw_grids` but I don't fully understand the `np.transpose` in there so I don't know how to do that. Do you think this is feasible?

Reply:
Good point. Could you elaborate what you mean by `raw_grids`? Maybe instead we could try a way to modify `FlatGrid` indexing such that the convolution output is sorted appropriately. I think this would confine our ordering scheme to be nested or analogous. But after all any form of afterwards sorting should be avoided if possible. Let me know what you think and I can have a look.